receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -77,6 +85,36 @@ func newMetricFamily(metricName string, mc MetadataCache) MetricFamily {
 	}
 }
 
+// Define manualy the metadata of prometheus scrapper internal metrics
+func defineInternalMetric(metricName string, metadata scrape.MetricMetadata, logger *zap.Logger) scrape.MetricMetadata {
+	if metadata.Metric != "" && metadata.Type != "" && metadata.Unit != "" && metadata.Help != "" {
+		logger.Debug("Internal metric seems already fully defined")
+		return metadata
+	}
+	metadata.Metric = metricName
+
+	switch metricName {
+	case scrapeUpMetricName:
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The scraping was sucessful"
+	case "scrape_duration_seconds":
+		metadata.Unit = "seconds"
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "Duration of the scrape"
+	case "scrape_samples_scraped":
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The number of samples the target exposed"
+	case "scrape_series_added":
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The approximate number of new series in this scrape"
+	case "scrape_samples_post_metric_relabeling":
+		metadata.Type = textparse.MetricTypeGauge
+		metadata.Help = "The number of samples remaining after metric relabeling was applied"
+	}
+	logger.Info("Internal metric defined", zap.String("metadata", fmt.Sprintf("%+v", metadata)))
```

Comment:
This line is probably not necessary.  It would be incredibly verbose, as it will be logged multiple times per-target per-scrape.